units/legacy_alerts.go

```diff
@@ -0,0 +1,150 @@
+package units
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/alerts"
+	"github.com/evergreen-ci/evergreen/model/alert"
+	"github.com/evergreen-ci/evergreen/model/user"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/mongodb/grip/sometimes"
+	"github.com/pkg/errors"
+)
+
+const legacyAlertRunnerJobName = "legacy-alert-runner"
+
+func init() {
+	registry.AddJobType(legacyAlertRunnerJobName,
+		func() amboy.Job { return makeLegacyAlertsJob() })
+
+}
+
+type legacyAlertsJob struct {
+	job.Base `bson:"job_base" json:"job_base" yaml:"job_base"`
+	env      evergreen.Environment
+}
+
+func makeLegacyAlertsJob() *legacyAlertsJob {
+	return &legacyAlertsJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    legacyAlertRunnerJobName,
+				Version: 0,
+			},
+		},
+	}
+}
+
+func NewLegacyAlertsRunnerJob(env evergreen.Environment, id string) amboy.Job {
+	j := makeLegacyMonitorRunnerJob()
+	j.env = env
+	j.SetID(fmt.Sprintf("%s.%s", legacyAlertRunnerJobName, id))
+	return j
+}
+
+func (j *legacyAlertsJob) Run(ctx context.Context) {
+	if j.env == nil {
+		j.env = evergreen.GetEnvironment()
+	}
+	config := j.env.Settings()
+	qp := alerts.NewQueueProcessor(config, evergreen.FindEvergreenHome())
+
+	flags, err := evergreen.GetServiceFlags()
+	if err != nil {
+		j.AddError(errors.Wrap(err, "error retrieving admin settings"))
+		return
+	}
+	if flags.AlertsDisabled {
+		grip.InfoWhen(sometimes.Percent(evergreen.DegradedLoggingPercent),
+			message.Fields{
+				"runner":  "alerts",
+				"message": "alerts are disabled, exiting",
+				"id":      j.ID(),
+			})
+		return
+	}
+
+	if len(config.SuperUsers) == 0 {
+		grip.Warning(message.Fields{
+			"message": "no superusers configured, some alerts may have no recipient",
+			"runner":  "alerts",
+			"id":      j.ID(),
+		})
+	}
+	superUsers, err := user.Find(user.ByIds(config.SuperUsers...))
+	if err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
+			"runner":    "alerts",
+			"status":    "failed",
+			"id":        j.ID(),
+			"operation": "finding super users",
+		}))
+
+		j.AddError(errors.Wrap(err, "problem getting superuser list"))
+		return
+	}
```

Comment:
What happens if superUsers is nil here? i.e., if all of the super users that were listed are invalid

Reply:
we would have logged a bunch at reasonably high levels. Nothing is really different here than it was in the old model.